packages/liveblocks-core/src/room.ts

```diff
@@ -2483,7 +2471,7 @@ export type InternalRoom<
   onVisibilityChange: (visibilityState: DocumentVisibilityState) => void;
 };
 
-export function createRoom<
+export function createRoomMachine<
```

Comment:
I'm on board with most of this renaming, but personally I don't think we should rename the room to roomMachine. I think it's fairly obvious now what is is through types, especially if you return a `RoomMachine` type.

Reply:
Sure, can change it back. There used to be a difference between "internal room" and "public room" with different APIs, think:

```tsx
const internalRoom = createRoom();

// The public Room API, as documented on
// https://liveblocks.io/docs/api-reference/liveblocks-client#Room
const room = internalRoom.room;
```

With this function name change, I wanted to stress that this function creates the _internal_ room type, aka the "RoomMachine".

But the more I'm refactoring this, though, the less of a difference there seems to be needed here. I think we may be able to ditch the idea of an "internal room" altogether and just return the (public) room API directly from this function.

I'll refactor this a bit more today, and see if I can remove the idea of the "internal room" (aka the "room machine") and instead just make the public room API. That will help to simplify the implementation a bit more internally.